dataprofiler/profilers/profile_builder.py

```diff
@@ -1402,17 +1419,46 @@ def tqdm(l):
         # Newly introduced features (python3.8) improves the situation
         sample_ids = np.array(sample_ids)
 
-        # Create structured profile objects
-        new_cols = set()
-        for col in data.columns:
-            if col not in self._profile:
-                self._profile[col] = StructuredColProfiler(
-                    sample_size=sample_size,
-                    min_true_samples=min_true_samples,
-                    sample_ids=sample_ids,
-                    options=self.options
-                )
-                new_cols.add(col)
+        # Create StructuredColProfilers (must be either first initialization
+        # or unique column names)
+        new_cols = False
+        if not duplicate_cols_present:
+            # Given duplicate columns despite duplicate columns not being
+            # present in nonempty _profile
+            if initialized and duplicate_cols_given:
+                raise ValueError("Attempted to update data with duplicate "
+                                 "column names that weren't present before "
+                                 "update. Schema must be identical when "
+                                 "profiling data with duplicate column names.")
```

Comment:
this should error / be raised prior to any calculations being done, including sample_ids

Reply:
I'm slightly confused by these conditionals. The case where duplicate columns should error is if duplicate columns are present and new data is added which has the same name as the duplicate column.

Reply:
However, I'm wondering if we should just require the schema to always match and not allow a profile to be updated a col at a time as this causes some headaches.